SimSim/Modules/Realm/Realm.m

```diff
@@ -10,37 +10,55 @@
 #import <Cocoa/Cocoa.h>
 #import "FileManager.h"
 
-#define PATH_REALM_FILES        @"Library/Caches"
+#define PATHS_REALM_FILES  [NSArray arrayWithObjects: @"Documents", @"Library/Caches", nil]
 #define REALM_APP_NAME          @"Realm Browser"
 #define REALM_APP_URL           @"http://itunes.apple.com/es/app/realm-browser/id1007457278"
 
 @implementation Realm
 
 
-- (NSMenu *)generateRealmMenuForPath:(NSString *)aPath {
-    
-    NSMenu* menuRealm = [[NSMenu alloc] initWithTitle:@"Realm Browser"];
-    [menuRealm setAutoenablesItems:NO];
+- (void)generateRealmMenuForPath:(NSString *)aPath forMenu:(NSMenu *)menu withHotKey:(NSNumber *)hotkey icon:(NSImage *)icon {
+
     NSArray *realmFiles = [[self class] findRealmFiles:aPath];
-    
+
+    if ([realmFiles count] == 0) { return; }  // Skip if there are no realm files
+
     BOOL isRealmBrowserInstalled = [[self class] isRealmBrowserAvailable];
-    
-    for (NSString *fileName in realmFiles) {
-        NSMenuItem* menuItem = [[NSMenuItem alloc] initWithTitle:fileName action:@selector(openRealmFile:) keyEquivalent:@""];
-        [menuItem setTarget:self];
-        [menuItem setRepresentedObject:[NSString stringWithFormat:@"%@/%@/%@", aPath, PATH_REALM_FILES, fileName]];
-        [menuItem setEnabled:isRealmBrowserInstalled];;
-        [menuRealm addItem:menuItem];
+    NSMenuItem* realmMenuItem = nil;
+
+    if (isRealmBrowserInstalled == false) { // There is at least one realm file but no realmbrowser installed
```

Comment:
I like your idea but I'm not sure if its needed, since most people won't see this more than once.